functions.test.js

```diff
@@ -0,0 +1,114 @@
+    expect(elementsToOccurrences(shuffledArray)).toEqual(
+      elementsToOccurrences(input)
+    );
+  });
+  test('["A"]', () => {
+    const input = ["A"];
+    const shuffledArray = shuffleArray([...input]);
+    // Expect shuffled array to have the same order of elements (1 element)
+    expect(shuffledArray).toEqual(input);
+    // Expect the count of each element be the same in both input and shuffledArray
+    expect(elementsToOccurrences(shuffledArray)).toEqual(
+      elementsToOccurrences(input)
+    );
+  });
+  test("[]", () => {
+    const input = [];
+    const shuffledArray = shuffleArray([...input]);
+    // Expect shuffled array to have the same order of elements (1 element)
+    expect(shuffledArray).toEqual(input);
+    // Expect the count of each element be the same in both input and shuffledArray
+    expect(elementsToOccurrences(shuffledArray)).toEqual(
+      elementsToOccurrences(input)
+    );
+  });
+});
+
+describe("Test createWordReadingGlossaryTriplets", () => {
+  const arrayFromJSON = [
+    { kanji: "火", "hiragana/katakana": "ひ", glossary: "fire" },
+    { kanji: "水", "hiragana/katakana": "みず", glossary: "water" },
+    { kanji: "木", "hiragana/katakana": "き", glossary: "wood" },
+    { kanji: "土", "hiragana/katakana": "ど", glossary: "earth" },
+    { kanji: "心臓", "hiragana/katakana": "しんぞう", glossary: "heart" },
+    { kanji: "空", "hiragana/katakana": "そら", glossary: "sky" },
+  ];
+  test("3 expected entries", () => {
+    expect(createWordReadingGlossaryTriplets(3, arrayFromJSON)).toEqual([
+      ["火", "ひ", "fire"],
+      ["水", "みず", "water"],
+      ["木", "き", "wood"],
+    ]);
+  });
+  test("4 expected entries", () => {
+    expect(createWordReadingGlossaryTriplets(4, arrayFromJSON)).toEqual([
+      ["火", "ひ", "fire"],
+      ["水", "みず", "water"],
+      ["木", "き", "wood"],
+      ["土", "ど", "earth"],
+    ]);
+  });
+  test("5 expected entries", () => {
+    expect(createWordReadingGlossaryTriplets(5, arrayFromJSON)).toEqual([
+      ["火", "ひ", "fire"],
+      ["水", "みず", "water"],
+      ["木", "き", "wood"],
+      ["土", "ど", "earth"],
+      ["心臓", "しんぞう", "heart"],
+    ]);
+  });
+  test("6 expected entries", () => {
+    expect(createWordReadingGlossaryTriplets(6, arrayFromJSON)).toEqual([
+      ["火", "ひ", "fire"],
+      ["水", "みず", "water"],
+      ["木", "き", "wood"],
+      ["土", "ど", "earth"],
+      ["心臓", "しんぞう", "heart"],
+      ["空", "そら", "sky"],
+    ]);
+  });
+  test("More expected entries than the length of arrayJSON", () => {
+    expect(createWordReadingGlossaryTriplets(7, arrayFromJSON)).toEqual([
+      ["火", "ひ", "fire"],
+      ["水", "みず", "water"],
+      ["木", "き", "wood"],
+      ["土", "ど", "earth"],
+      ["心臓", "しんぞう", "heart"],
+      ["空", "そら", "sky"],
+    ]);
+  });
+});
+
+describe("Test formatTime", () => {
+  test("More than 9", () => {
+    expect(formatTime(10)).toBe(10);
+    expect(formatTime(31)).toBe(31);
+    expect(formatTime(54)).toBe(54);
+  });
```

Comment:
There is a special syntax in `jest` that streamlines writing this type of simple tests. It makes use of `.each` method:
```javascript
test.each([10, 31, 54])("More than 9: %i", (value) => {
  expect(formatTime(value)).toBe(value);
});

// OR more explicit format using backticks ``

test.each`
  value
  ${10}
  ${31}
  ${54}
`("More than 9: $value", ({ value }) => {
  expect(formatTime(value)).toBe(value);
});
```
in the first variation you can put whatever JS objects you want (dicts too) which can be handy. In the second "table" syntax you can have many columns, which is often more readable. Here is an example from another project:
```javascript
 test.each`
    code      | index | expected
    ${''}     | ${1}  | ${''}
    ${'abcd'} | ${0}  | ${'a'}
    ${'abcd'} | ${1}  | ${'b'}
    ${'abcd'} | ${2}  | ${'c'}
    ${'abcd'} | ${3}  | ${'d'}
    ${'abcd'} | ${4}  | ${''}
  `('inputValue returns correct value', ({ code, index, expected }) => {
    const wrapper = shallowMountComponent();
    wrapper.setData({ code });
   
    const result = wrapper.vm.inputValue(index);
    
    expect(result).toBe(expected);
  });
  ```